fadecandy_driver/src/ros/fadecandy_driver_ros.cpp

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright (c) 2021 Eurotec
+//
+
+#include "./fadecandy_driver_ros.h"
+#include <fadecandy_msgs/LEDArray.h>
+
+namespace fadecandy_driver
+{
+FadecandyDriverRos::FadecandyDriverRos()
+{
+  ros::NodeHandle nh;
+
+  timer_ = nh.createTimer(ros::Duration(1.), &FadecandyDriverRos::timerCallback, this);
+
+  diagnostic_updater_.add("Info", this, &FadecandyDriverRos::diagnosticsCallback);
+
+  led_subscriber = nh.subscribe<fadecandy_msgs::LEDArray>("/set_leds", 1, &FadecandyDriverRos::setLedsCallback, this);
+  initialized_ = false;
+}
+
+void FadecandyDriverRos::run()
+{
+  if (fadecandy_device_ != NULL)
+  {
+    release();
+    fadecandy_device_ = NULL;
+  }
+
+  if (!initialized_)
+  {
+    initialized_ = FadecandyDriverRos::intialize();
+    if (initialized_)
+    {
+      ROS_INFO("Connected to Fadecandy device");
+
+      diagnostic_updater_.setHardwareID(serial_number_);
+    }
+    else
+    {
+      throw std::runtime_error("Device not found! Could not write on the driver.");
+    }
+  }
+}
+
+void FadecandyDriverRos::setLedsCallback(const fadecandy_msgs::LEDArrayConstPtr& led_array_msg)
+{
+  std::vector<std::vector<Color>> led_array_colors;
+  std::vector<Color> led_strip_colors;
```

Comment:
you don't need this variable

Reply:
This variable is needed because `led_array_colors` is a vector of vectors and in the for loop the `led_array_colors`  vector is built after many iterations and then would be added to the main vector `led_array_colors`